mmtk/src/lib.rs

```diff
@@ -119,23 +121,18 @@ impl VMBinding for OpenJDK {
     type VMReferenceGlue = reference_glue::VMReferenceGlue;
 }
 
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+
+pub static MMTK_INITIALIZED: AtomicBool = AtomicBool::new(false);
+
 lazy_static! {
+    pub static ref BUILDER: MMTKBuilder = MMTKBuilder::new();
     pub static ref SINGLETON: MMTK<OpenJDK> = {
-        #[cfg(feature = "nogc")]
-        std::env::set_var("MMTK_PLAN", "NoGC");
-        #[cfg(feature = "semispace")]
-        std::env::set_var("MMTK_PLAN", "SemiSpace");
-        #[cfg(feature = "gencopy")]
-        std::env::set_var("MMTK_PLAN", "GenCopy");
-        #[cfg(feature = "marksweep")]
-        std::env::set_var("MMTK_PLAN", "MarkSweep");
-        #[cfg(feature = "markcompact")]
-        std::env::set_var("MMTK_PLAN", "MarkCompact");
-        #[cfg(feature = "pageprotect")]
-        std::env::set_var("MMTK_PLAN", "PageProtect");
-        #[cfg(feature = "immix")]
-        std::env::set_var("MMTK_PLAN", "Immix");
-        MMTK::new()
+        assert!(!MMTK_INITIALIZED.load(Ordering::Relaxed));
+        let ret = mmtk::memory_manager::gc_init(&BUILDER);
+        MMTK_INITIALIZED.store(true, std::sync::atomic::Ordering::Relaxed);
```

Comment:
~~This should be `Release` (or `SeqCst` if we want to be more conservative).~~

`Relaxed` should be is OK here.